app/src/main/java/org/mozilla/focus/settings/ManualAddSearchEngineSettingsFragment.java

```diff
@@ -136,9 +164,13 @@ protected Boolean doInBackground(final Void... voids) {
         protected void onPostExecute(final Boolean isValidSearchQuery) {
             super.onPostExecute(isValidSearchQuery);
 
+            if (isCancelled()) {
```

Comment:
> Documentation says that a cancelled AsyncTask should never go to onPostExecute

Good to know.

> (though who knows if it can be called while in isCancelled! I'm going to ignore that case.)

I assume it can be, however:
- From the docs you pointed out, it sounds like the intention of cancelling is ending the long running background task, rather than stopping the (hopefully) short UI thread operation
- In this case, I call cancel from the UI thread so it can't happen inside onPostExecute.

> So basically calling AsyncTask.cancel only sets a flag, and you have to check the flag yourself. 

Not always – you can tell AsyncTask.cancel to interrupt the thread (which stops execution and throws an InterruptedException) and which I did here

> And at this point, since this call is only 4s long anyway...

I don't think there's a way to stop a blocking call anyway, beyond the interrupt method mentioned above.

Reply:
> Not always – you can tell AsyncTask.cancel to interrupt the thread (which stops execution and throws an InterruptedException) and which I did here

Although note that the thread is not stopped magically (with an InterruptedException). Code inside the Thread might need to check for the interrupted state to stop running _early_. I'm not sure if HttpUrlConnection actually does that (like for example sleep()).

https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html